src/main/java/command/AddDeadlineCommand.java

```diff
@@ -0,0 +1,31 @@
+package command;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import task.*;
+import duke.*;
+
+public class AddDeadlineCommand extends AddCommand {
+
+    private String desc;
+    private LocalDate by;
+
+    public AddDeadlineCommand(String desc, String by) {
+        this.desc = desc;
+        this.by = LocalDate.parse(by);
+    }
+
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
+        // tasks
+        Deadline newDeadline = new Deadline(desc, by);
+        tasks.add(newDeadline);
+        // ui
+        String response = respond(newDeadline, tasks.size());
+        String result = ui.showResponse(response);
+        // storage
```

Comment:
Would it better to remove the comments? Your code is written very clear, and it is easy to understand. This would adhere better to the [conventions for commenting](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#do-not-repeat-the-obvious).